alica_dummy_proxy/src/communication/AlicaDummyCommunication.cpp

```diff
@@ -172,7 +172,7 @@ template <>
 void InProcQueue<alica::AgentAnnouncement>::process(std::unique_ptr<alica::AgentAnnouncement>& first)
 {
     std::vector<AlicaDummyCommunication*> registeredModules = _commModules.getModules();
-    essentials::IdentifierConstPtr prev = first->senderID;
+    uint64_t prev = first->senderID;
```

Comment:
One suggestion, define alica::AgentId as uint64_t  at a single place and use AgentId everywhere rather than directly using uint64_t. If in future we want to change the type of AgentId, we can do it at a single place rather than changing all over the engine again.